### PR TITLE
feat: support `/` in branch name

### PR DESCRIPTION
**Pull request by xcv58**
You can preview it here:

https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s/blob/index/html/test/src/vs/test.txt

https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s/blob/HEAD/src/vs/code/browser/workbench/workbench.ts

https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s/blob/ef7c21ce7d21eb700190569753178369fc161663/src/vs/test.txt

There is a bug that prevents show binary files:
image
It's probably related to https://github.com/microsoft/vscode/issues/112641

I tried to upgrade the VS Code to 1.53.x https://github.com/microsoft/vscode/releases but there are changes break the app. 

@conwnet do you know the effort to upgrade VS Code to the latest version?

### CODE REVIEW

File: src/vs/code/browser/workbench/workbench.ts [hunk 1]
```diff
@@ -401,11 +401,15 @@ class WindowIndicator implements IWindowIndicator {
 (function () {
 	const route = parseGitHubUrl(window.location.href);
 	const config: IWorkbenchConstructionOptions & { folderUri?: UriComponents, workspaceUri?: UriComponents } = {
-		folderUri: URI.from({ scheme: "github1s", path: '/', authority: `${route.owner}+${route.repo}+${route.branch}` }),
+		/**
+		 * We couldn't figure out the correct branch here since branch might has `/` i.e. `feat/new-feat-1`.
+		 * So we have to use the `route.path` in authority and figure out branch in github1s extension.
+		 */
+		folderUri: URI.from({ scheme: "github1s", path: '/', authority: `${route.owner}+${route.repo}+${route.path}` }),
```
Comment (xcv58) [inline src/vs/code/browser/workbench/workbench.ts:408]: @conwnet could you please confirm whether this assumption is correct?

The code would be significantly simpler if we could call `commandService.executeCommand('github1s.get-github-branches', window.location.href)` here.

Comment (conwnet) [reply src/vs/code/browser/workbench/workbench.ts:408]: yes, but we can't call `commandService` here, I want to remove the `branch` part from URI authority, and deal with `branch` in github1s extension, let me have a try

Comment (xcv58) [reply src/vs/code/browser/workbench/workbench.ts:408]: I did try this direction. The only missing part is that we use `path = ‘/‘` to indicate the repo’s root which will lose the original pathname. A possible solution is using query or fragment in URI to indicate the information.

Comment (conwnet) [reply src/vs/code/browser/workbench/workbench.ts:408]: yes, after reading the documentation of [VS Code Extension Source Control](https://code.visualstudio.com/api/extension-guides/scm-provider), I realized the importantce for **Exact URI** of resouce. I adjusted that

Comment (conwnet) [reply src/vs/code/browser/workbench/workbench.ts:408]: change the `authority` as format `{owner}/{repo}/{ref}`, we assume the empty `authority` means github1s should get it from `window.location.href` https://github.com/xcv58/github1s/pull/6

**Review by Siddhant-K-code — APPROVED** [review]

**Comment by xcv58** [comment]
It works for all kind of URLs

- No path
https://github1s-git-branches-command.xcv58.vercel.app

- Multiple parts branch
https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s/blob/index/html/test/src/vs/test.txt

- Repo only
https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s

- Commit id
https://github1s-git-branches-command.xcv58.vercel.app/xcv58/github1s/blob/ef7c21ce7d21eb700190569753178369fc161663/src/vs/test.txt

- tag-based URL
https://github1s-git-branches-command.xcv58.vercel.app/xcv58/Tab-Manager-v2/blob/v1.14.2/packages/extension/package.json

**Pull request merged**